modules/govuk/manifests/app/nginx_vhost.pp

```diff
@@ -94,11 +94,19 @@
     $nginx_extra_config_real = $nginx_extra_config
   }
 
+  # Force HTTP basic auth for the training environment, since it doesn't have
```

Comment:
Instead of using if - else for a specific facter inside a generic class, I would override the parameter vhost_protected in Hiera, for instance:
hieradata/development.yaml:govuk::apps::whitehall::vhost_protected: false
inside hieradata/class/training.yaml with true